module/docs/entra-powershell-v1.0/Microsoft.Graph.Entra/New-EntraServiceAppRoleAssignment.md

```diff
@@ -106,46 +107,108 @@ DeletedDateTime Id                                          AppRoleId
 
 - `ObjectId`:  The ObjectId of the app's service principal.
-- `ResourceId`: The ObjectId of the app's service principal.
-- `Id`: The Id of the app role (defined on the app's service principal) to assign to the user. If no app roles are defined to the resource app, you can use `00000000-0000-0000-0000-000000000000` to indicate that the app is assigned to the user.
-- `PrincipalId`: The ObjectId of the user to which you're assigning the app role.
+- `ResourceId`: The `Id` of the resource servicePrincipal (the API) that defines the app role (the application permission).
+- `Id`: The `Id` of the app role (defined on the resource service principal) to assign to the client service principal. If no app roles are defined on the resource app, you can use `00000000-0000-0000-0000-000000000000`.
+- `PrincipalId`: The `Id` of the client service principal to which you're assigning the app role.
+
+### Example 4: Assign permissions to a service principal
+
+```powershell
+Connect-Entra -Scopes 'AppRoleAssignment.ReadWrite.All'
+
+$ServicePrincipal = Get-EntraServicePrincipal -Filter "DisplayName eq 'Contos App 1'"
+$FilterParams = @{
+    Filter = "AppId eq '00000003-0000-0000-c000-000000000000'"
+}
+
+# Get Graph App
+$GraphApp = Get-EntraServicePrincipal @FilterParams
+
+# Get App Role
+$AppRole = $GraphApp.AppRoles | Where-Object { $_.Value -eq 'User.Read.All' }
+
+# Assign the permission
+$Params = @{
+    PrincipalId = $ServicePrincipal.Id
+    ResourceId  = $GraphApp.Id
+    Id          = $AppRole.Id
+    ObjectId    = $ServicePrincipal.Id
+}
+
+New-EntraServiceAppRoleAssignment @Params
+```
+
+This example shows how to assign application permission to a service principal to support your automation scenarios.
+
+### Example 5: Assign multiple permissions to a service principal
+
+```powershell
+Connect-Entra -Scopes 'AppRoleAssignment.ReadWrite.All'
+
+$GetPermissions = 'Application.Read.All', 'User.Read.All'
+# Get service principal
+$ServicePrincipal = Get-EntraServicePrincipal -Filter "DisplayName eq 'My Service Principal'"
+
+# Get Graph App Id
+$GraphAppFilterParams = @{
+    Filter = "AppId eq '00000003-0000-0000-c000-000000000000'"
+}
+$GraphApp = (Get-EntraServicePrincipal @GraphAppFilterParams)
+
+# Get App Roles
+$GraphAppRoles = $GraphApp.AppRoles | Where-Object { $_.Value -in $GetPermissions }
+
+# Assign the permission to App
+foreach ($AppRole in $GraphAppRoles) {
+    $Params = @{
+        PrincipalId = $ServicePrincipal.Id
+        ResourceId  = $GraphApp.Id
+        Id          = $AppRole.Id
+        ObjectId    = $ServicePrincipal.Id
+    }
+    
+    New-EntraServiceAppRoleAssignment @Params
+}
+```
+
+This example shows how to assign application permissions to a service principal to support your automation scenarios.
 
 ## Parameters
 
 ### -Id
 
-Specifies the ID.
+Specifies the `Id` of the app role (defined on the resource service principal) to assign to the client service principal.
```

Comment:
Because the principal can be a user, group or service principal.
```suggestion
Specifies the `Id` of the app role (defined on the resource service principal) to assign to the principal.
```